pystiche/misc.py

```diff
@@ -280,16 +280,13 @@ def reduce(x: torch.Tensor, reduction: str) -> torch.Tensor:
 
 
 @contextlib.contextmanager
-def suppress_warnings(*categories: Type[Warning]) -> Iterator[None]:
+def suppress_deprecation_warnings(*categories: Type[Warning]) -> Iterator[None]:
     if not categories:
         categories = (UserWarning,)
-    old_filters = set(warnings.filters)  # type: ignore[attr-defined]
     for category in categories:
-        warnings.filterwarnings("ignore", category=category)
-    new_filters = set(warnings.filters) - old_filters  # type: ignore[attr-defined]
-    try:
-        yield
-    finally:
-        for filter in new_filters:
-            warnings.filters.remove(filter)  # type: ignore[attr-defined]
-        warnings._filters_mutated()  # type: ignore[attr-defined]
+        warnings.filterwarnings(
```

Comment:
I think just doing `filterwarnings` will set them "forever". The old version had the ability to reset these filters after the context manager is done. I think it would be even better to use [`warnings.catch_warnings`](https://docs.python.org/3/library/warnings.html#warnings.catch_warnings) if that is possible.

Reply:
You are right, for some reason - using context manager wasn't working here, though using only decorator works. I've pushed this in the recent commit.